plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobProgressBar.qml

```diff
@@ -82,54 +29,85 @@ Item
-                            "resuming",
-                            "wait_cleanup"
-                        ]
-                        if (inactiveStates.indexOf(state) > -1 && remainingTime > 0)
-                        {
-                            return UM.Theme.getColor("monitor_progress_fill_inactive")
-                        }
-                    }
-                    return "#0a0850" // TODO: Theme!
-                }
+                color: printJob && printJob.isActive ? "#0a0850" : "#9392b2" // TODO: Theme!
                 radius: 4 * screenScaleFactor // TODO: Theme!
             }
         }
     }
     Label
     {
-        id: progressLabel
+        id: percentLabel
         anchors
         {
             left: progressBar.right
             leftMargin: 18 * screenScaleFactor // TODO: Theme!
         }
-        text: progressText
-        color: "#374355" // TODO: Theme!
+        text: Math.round(printJob.progress * 100) + "%"
+        color: printJob && printJob.isActive ? "#374355" : "#babac1" // TODO: Theme!
         width: contentWidth
         font: UM.Theme.getFont("medium") // 14pt, regular
 
+        // FIXED-LINE-HEIGHT:
+        height: 18 * screenScaleFactor // TODO: Theme!
+        verticalAlignment: Text.AlignVCenter
+    }
+    Label
+    {
+        id: statusLabel
+        anchors
+        {
+            left: percentLabel.right
+            leftMargin: 18 * screenScaleFactor // TODO: Theme!
+        }
+        color: "#374355" // TODO: Theme!
+        font: UM.Theme.getFont("medium") // 14pt, regular
+        text:
+        {
+            if (!printJob)
+            {
+                return "";
+            }
+            switch (printJob.state)
+            {
+                case "wait_cleanup":
+                    if (printJob.timeTotal > printJob.timeElapsed)
+                    {
+                        return catalog.i18nc("@label:status", "Aborted")
+                    }
+                    return catalog.i18nc("@label:status", "Finished")
+                case "sent_to_printer":
+                    return catalog.i18nc("@label:status", "Preparing...")
+                case "aborting":
+                    return catalog.i18nc("@label:status", "Aborting...")
+                case "aborted":
+                    return catalog.i18nc("@label:status", "Aborted")
+                case "pausing":
+                    return catalog.i18nc("@label:status", "Pausing...")
+                case "paused":
+                    return catalog.i18nc("@label:status", "Paused")
+                case "resuming":
+                    return catalog.i18nc("@label:status", "Resuming...")
+                case "queued":
+                    return catalog.i18nc("@label:status", "Action required")
+                default:
+                    return catalog.i18nc("@label:status", "Finishes ") + OutputDevice.getDateCompleted( printJob.timeRemaining ) + " at " + OutputDevice.getTimeCompleted( printJob.timeRemaining )
```

Comment:
This string building needs to be replaced with 1 translatable string/template.

Reply:
Maybe it is now time to move some of stuff to the Python side? The Python code is actually full of i18n strings too.